fastparquet/writer.py

```diff
@@ -1170,13 +1169,8 @@ def merge(file_list, verify_schema=True, open_with=default_open,
     -------
     ParquetFile instance corresponding to the merged data.
     """
-    basepath, fmd = metadata_from_many(file_list, verify_schema, open_with,
-                                       root=root)
-
-    out_file = join_path(basepath, '_metadata')
-    write_common_metadata(out_file, fmd, open_with, no_row_groups=False)
-    out = api.ParquetFile(out_file, open_with=open_with)
-
-    out_file = join_path(basepath, '_common_metadata')
-    write_common_metadata(out_file, fmd, open_with)
+    out = api.ParquetFile(file_list, verify_schema, open_with, root)
```

Comment:
I would be tempted to keep the original here, since creating the `ParquetFile` involves extra work (in `._set_attrs`) beyond merging the row-group metadata, such as processing directory names, which might fail.

Reply:
Thanks for the feedbacks @martindurant . I clearly do not pretend to master fastparquet codebase and I may very likely miss some stuffs, so please I beg your pardon if the following is erroneous. This is the state of my understanding.

> creating the ParquetFile involves extra work (in ._set_attrs) beyond merging the row-group metadata
 
I am not sure the point 'extra work' is valid here as a `ParquetFile` instance is created in the current code few lines later, [line 1181](https://github.com/dask/fastparquet/blob/c6b6a76fa480304aac79aa13d32db40da44155dc/fastparquet/writer.py#L1181):
```python
    out = api.ParquetFile(out_file, open_with=open_with)
```
So `.set_attrs` is already executed in current codebase as far as I understand.

> such as processing directory names, which might fail.

Here, this is where I am the less at ease with what I say. What I can see:
- in current codebase, `.set_attrs` is already executed, so it would fail as well wouldn't it?
- I have (quickly) reviewed the code contained in `.set_attrs` and the other methods it calls `._read_partitions()` and `._dtypes()`  (skipping the other stuffs related to `fmd` and `row_groups` that are generated in both cases - current code and my proposal - by `metadata_from_many()` so for which I identify no reason any changes would occur). In this code (contained in `.set_attrs` and the other methods it calls `._read_partitions()` and `._dtypes()`), I see no `raise xxxError`. So I miss the reason why it would fail.

I will not insist on keeping this part of the proposal, this is not the point of the set of 3 PRs I am submitting.
But I am also uncertain about how your feedback applies.
Please, given what I could see, do you confirm it is best to revert this part of the code?
Thanks in advance for your reply. I will proceed the revert as soon as you confirm it.

Bests,

Reply:
OK, I see you are right. The logic is reversed, but ought to amount to the same thing.